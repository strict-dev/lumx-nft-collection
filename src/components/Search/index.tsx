import SearchIcon from '@public/assets/search.svg'
import Image from 'next/image'
import { useRef, useState } from 'react'

const Search = () => {
  const [isSearchOpen, setIsSearchOpen] = useState(false)
  const inputRef = useRef<HTMLInputElement>(null)

  const handleSearch = () => {
    setIsSearchOpen(!isSearchOpen)
    inputRef.current?.focus()
  }

  return (
    <button
      className={`flex items-center px-3 py-1 rounded-sm transition
      ${isSearchOpen ? 'bg-neutral-900' : ''}
      focus-within:outline focus-within:outline-neutral-700`}
    >
      <Image
        src={SearchIcon}
        alt="Search"
        width={24}
        height={24}
        onClick={handleSearch}
        layout="fixed"
      />

      <input
        onBlur={() => setIsSearchOpen(false)}
        ref={inputRef}
        type="text"
        placeholder="Search"
        className={`px-3 bg-transparent w-36 outline-none transition : ;
        ${isSearchOpen ? '' : 'hidden'}`}
      />
    </button>
  )
}

export default Search
