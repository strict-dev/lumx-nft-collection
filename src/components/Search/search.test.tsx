import { render } from '@testing-library/react'

import Search from '.'

describe('<Search />', () => {
  it('should render the heading', () => {
    const { container } = render(<Search />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
