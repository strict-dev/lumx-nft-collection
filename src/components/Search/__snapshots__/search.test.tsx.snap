// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render the heading 1`] = `
<button
  class="flex items-center px-3 py-1 rounded-sm transition
      
      focus-within:outline focus-within:outline-neutral-700"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 24px; height: 24px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
  >
    <img
      alt="Search"
      data-nimg="fixed"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <input
    class="px-3 bg-transparent w-36 outline-none transition : ;
        hidden"
    placeholder="Search"
    type="text"
  />
</button>
`;
